From cfe5cd8009aa0fb225c817b6b89d9dedd3316bb2 Mon Sep 17 00:00:00 2001
From: Kris Van Hees <kris.van.hees@oracle.com>
Date: Fri, 6 Feb 2026 12:50:49 -0500
Subject: [PATCH] pid: USDT probes cannot have "-" as function name

Signed-off-by: Kris Van Hees <kris.van.hees@oracle.com>
Reviewed-by: Eugene Loh <eugene.loh@oracle.com>
---
 libdtrace/dt_pid.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libdtrace/dt_pid.c b/libdtrace/dt_pid.c
index a379a0fe..ed329b9d 100644
--- a/libdtrace/dt_pid.c
+++ b/libdtrace/dt_pid.c
@@ -1565,6 +1565,10 @@ dt_pid_create_usdt_probes(dtrace_probedesc_t *pdp, dtrace_hdl_t *dtp)
 	char *globpat = NULL;
 	int err = 0, i, nmatches = 0;
 
+	/* If the function name is "-", we're done. */
+	if (pdp->fun[0] == '-' && pdp->fun[1] == '\0')
+		return 0;
+
 	/* If it cannot end with a pid, we're done. */
 	if (pdp->prv[0] != '\0') {
 		char lastchar = pdp->prv[strlen(pdp->prv) - 1];
