diff --git a/test/unittest/actions/raise/tst.raise3.d b/test/unittest/actions/raise/tst.raise3.d
index 3c3fd20..ef09a8a 100644
--- a/test/unittest/actions/raise/tst.raise3.d
+++ b/test/unittest/actions/raise/tst.raise3.d
@@ -24,8 +24,8 @@ BEGIN
 {
 	/*
-	 * Wait no more than five seconds for the process to call ioctl().
+	 * Wait no more than fifteen seconds for the process to call ioctl().
 	 */
-	timeout = timestamp + 5000000000;
+	timeout = timestamp + 15000000000;
 }
 
 syscall::ioctl:entry
@@ -36,9 +36,9 @@ syscall::ioctl:entry
 {
 	raise(SIGINT);
 	/*
-	 * Wait no more than three seconds for the process to die.
+	 * Wait no more than ten seconds for the process to die.
 	 */
-	timeout = timestamp + 3000000000;
+	timeout = timestamp + 10000000000;
 }
 
 syscall::exit_group:entry
